controllers/components/kserve/kserve_controller.go

```diff
@@ -155,6 +144,7 @@ func (s *componentHandler) NewComponentReconciler(ctx context.Context, mgr ctrl.
 		WithAction(checkPreConditions).
 		WithAction(initialize).
 		WithAction(devFlags).
+		WithAction(removeLegacyFeatureTrackerOwnerRef).
```

Comment:
should this be moved into upgrade not in kserve?

Reply:
Removing owner reference is a little bit tricky as if it may lead to some resources hanging around, so I've implemented it as closed as when the resources are actually being replaced.